### DISCUSSION TITLE
GPS NMEA Output

### DISCUSSION THREAD

**Discussion by remcoderksen** [XCSoar/XCSoar#1407] [Ideas]
Hello,

My Trig TT21 transponder accepts GPS related NMEA 0183 sentences over RS-232 on connector pin 5 and with this I can enable ADS-B functionality on it. Now instead of cramming another power consuming GPS dongle, 12V>5V regulator, toggle-switch and fuse in my dashboard, I was thinking if I could use/proxy the GPS data that XCSoar gets from my FLARM Classic.

So this afternoon I wrote a simple driver that does exactly that job. It only outputs GPRMC, GPGGA and GPGSA sensences. 
With this output the idea is to connect the TX of a free TTY on the back of my OpenVario to my Trig TT21 pin 5 RX.

I was wondering about your opinion on this. Does it sounds plausible? 

Kind regards,

Remco

**Comment by lordfolken** [comment]
Really not a good idea. Please don´t. 

Transponders are life saving equipment that TCAS systems rely on to function properly.  Don't mess with it.  There is a reason they are certified equipment. Just like its GPS source has to be.

You need a certified gps signal for ads-b output. This one includes a signal quality field that is transmitted on ads-b. If that field is missing, many ATC Radar displays will filter your ads-b signal.

XCSoar is for Informational use _only_. Its not certified for any purpose. (Do i really need to add a disclaimer?)
XCsoar might crash, cause latency, be blocked and delay your gps stream or mangle it. 
XCSoar isn´t  written with real time guarantees in mind.  (Same reason it's not a flarm display replacement, or an Artificial Horizon replacement)

Your transponder is certified. Hooking uncertified equipment to your transponder will probably not go down well upon inspection and is probably only allowed with special precautions under EASA Part M.

**Reply by remcoderksen** [reply]
Hi!
Yes, you are totally right. I was still in the poc state and caught in the fun of programming and technical part. I would have come to the certified part eventually and would have stranded there with this same conclusion about the true reason of ADS-B. So I will abort this idea. The good part is that it taught me about the drivers and the parts of the code in XCSoar that I did not see before.

**Comment by ballfire** [comment]
You might want to check CS STAN CS-SC005b

It's possible to provide adsb out without an approved gnss source. 

Latency and non-real-time thing is a valid concern, though

**Comment by groundhogxc** [comment]
Yes. There is not a legality issue here. All the ADS-B sets you can buy below 1000.- are not certified, but legal.

Also, now and in the near future, ADS-B data is NOT used by the TCAS systems. And if it were, those devices would for sure evaluate the "certified GPS source" flag anyway, which would not be set in our case. 

Of course, there might be other informal (non-TCAS, and non-certified) implementations of some kind of collision warning based on ADS-B transmissions in some flight deck computers which ignore the flag. 
I am not aware of such a system at this point, but in that scenario the data latency and potential corruption could indeed make a difference. 

At any rate, I would always prefer to go directly to the data stream, split off diretly from the FLARM. Not sure what the TTRIG will do when it gets the full FLARM stream, however. Theoretically, it could ignore all the non-positional messages. But  I heard some transponders do not like that, in particular the older TQ's have apparently some issues in that case, according to old internet discussions. Which is supported by the fact that TQ offers a special modified GPS mouse (a completely normal consumer GPS), where the modification means that it is only emitting the most basic sentences.
Just try it out... Might have to check the baud rate compatibility.

**Comment by Scumi** [comment]
For info/reference: NMEA forwarding of GPS related sentences exists in the driver for the AirControlDisplay, mainly for the radio station database. See #1117.

**Comment by groundhogxc** [comment]
Actually, I almost forgot what I wanted to write in the first place - I actually think such a driver *is * useful. I looked for something like that myself a long time ago. (I would not want to connect a transponder that way, though)

In that case, I needed GPS data on another device, and I would have wanted to use such a driver not to a TTY, but to an IP address or a bluetooth connection. I think that can be very useful, but these are rare circumstances.

But at the time, when I was thinking about that, I had another use case for this: To run a second instance of XCSoar as backup on the smartphone, and use an NMEA stream (with GPS, but potentially even more info) from the main device to the secondary (smartphone) in order to keep that in sync. When the primary device fails, you could then continue by using the internal GPS with lower precision, and higher battery usage, and you would have the full trace in the secondary, including the wind information and whatever else your primary device records from the various senors on the plane. It could even synchronize MC values and ballast state and stuff like that.
Just as an idea...

(and lust but not least, streaming the NMEA out could be useful for debugging things, I guess)
